dbt_coves/tasks/generate/metadata.py

```diff
@@ -0,0 +1,280 @@
+            for csvdict in dicts_to_write:
+                if existing_rows:
+                    if csvdict not in existing_rows:
+                        writer.writerow(csvdict)
+                else:
+                    writer.writerow(csvdict)
+        console.print(
+            f"[green]{relation.name}[/green] metadata written to [green]{destination.absolute()}[/green]"
+        )
+
+    def get_nested_keys(self, json_cols, relation):
+        config_db = self.get_config_value("database")
+        if config_db:
+            config_db += "."
+        else:
+            config_db = ""
+        _, data = self.adapter.execute(
+            f"SELECT {', '.join(json_cols)} FROM {config_db}{relation.schema}.{relation.name} limit 1",
+            fetch=True,
+        )
+        result = dict()
+        if len(data.rows) > 0:
+            for idx, json_col in enumerate(json_cols):
+                value = data.columns[idx]
+                try:
+                    nested_key_names = list(json.loads(value[0]).keys())
+                    result[json_col] = {}
+                    for key_name in nested_key_names:
+                        result[json_col][key_name] = self.get_default_metadata_item(key_name)
+                except TypeError:
+                    console.print(
+                        f"Column {json_col} in relation {relation.name} contains invalid JSON.\n"
+                    )
+        return result
+
+    def get_templates_context(self, relation, columns, json_cols=None):
+        context = {
+            "relation": relation,
+            "columns": columns,
+        }
+        if json_cols:
+            context["nested"] = self.get_nested_keys(json_cols, relation)
+        return context
+
+    def get_existing_csv_rows(self, destination):
+        try:
+            with open(destination, newline="") as csvfile:
+                reader = csv.DictReader(csvfile, fieldnames=self.METADATA_HEADERS)
+                return [row for row in reader]
+        except FileNotFoundError:
+            return []
+
+    def generate(self, rels):
+        destination = self.get_config_value("destination")
+        options = {
+            "append_all": None,
+            "flatten_all": None,
+            "created_file": False,
+        }
+
+        for rel in rels:
+            csv_dest = self.generate_template(destination, rel)
+            csv_path = Path().joinpath(csv_dest)
+            existing_rows = self.get_existing_csv_rows(csv_path)
+            if not options["append_all"]:
+                if csv_path.exists() and existing_rows:
+                    console.print(f"[yellow]{csv_path.absolute()}[/yellow] exists.")
+                    append = questionary.select(
+                        f"Would you like to add {rel.name.lower()} metadata?",
+                        choices=[
+                            "Recreate file",
+                            "Add missing columns",
+                            "Add for all relations",
```

Comment:
This option is confusing.

Reply:
@ssassi `add for all relations` is appending to all relations, and not asking again. That's why it toggles `append_all`
should I remove it? And let `Add missing columns` append all csv rows for all relations

another detail I found: when using `Recreate file` shouldn't we presume an `append_all` behavior?